WordPress/src/main/java/org/wordpress/android/ui/posts/FeaturedImageHelper.kt

```diff
@@ -49,39 +55,34 @@ internal class FeaturedImageHelper @Inject constructor(
     }
 
     fun retryFeaturedImageUpload(
-        context: Context,
         site: SiteModel,
         post: PostModel
     ): MediaModel? {
         val mediaModel = getFailedFeaturedImageUpload(post)
         if (mediaModel != null) {
-            UploadService.cancelFinalNotification(context, post)
-            UploadService.cancelFinalNotificationForMedia(context, site)
+            uploadServiceFacade.cancelFinalNotification(post)
+            uploadServiceFacade.cancelFinalNotificationForMedia(site)
             mediaModel.setUploadState(MediaUploadState.QUEUED)
             dispatcher.dispatch(MediaActionBuilder.newUpdateMediaAction(mediaModel))
-            startUploadService(context, mediaModel)
+            startUploadService(mediaModel)
         }
         return mediaModel
     }
 
-    private fun startUploadService(context: Context, media: MediaModel) {
+    private fun startUploadService(media: MediaModel) {
         val mediaList = ArrayList<MediaModel>()
         mediaList.add(media)
-        UploadService.uploadMedia(context, mediaList)
+        uploadServiceFacade.uploadMedia(mediaList)
     }
 
     fun queueFeaturedImageForUpload(
-        context: Context,
         localPostId: Int,
         site: SiteModel,
         uri: Uri,
         mimeType: String?
-    ) {
-        val media = FluxCUtils.mediaModelFromLocalUri(context, uri, mimeType, mediaStore, site.id)
-        if (media == null) {
-            ToastUtils.showToast(context, R.string.file_not_found, ToastUtils.Duration.SHORT)
-            return
-        }
+    ): SnackbarMessageHolder? {
```

Comment:
>Should this method return a ToastMessageHolder instead?

Ohh, I forgot to change that when I replaced the Snackbar with a Toast.

>After reviewing what the method does, I'd actually prefer to return a Boolean here for success/failure.

I was considering that as well but I felt like the method can fail from various reasons. However, I agree that it should still probably return an enum or something like that. Having said that, for the sake of simplicity and since there is only one possible error scenario, I refactored the code and it returns boolean now. (https://github.com/wordpress-mobile/WordPress-Android/pull/10662/commits/5c9f4a7210f3501b5e2d76c12fd3de8f87b069d3)

Reply:
An `enum` would be even better 👍